*: reorganize content to display version switcher

lilin90 opened the pull request:
<!--Thanks for your contribution to TiDB documentation. See [CONTRIBUTING](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md) before filing this PR.-->

### What is changed, added or deleted? <!--Required-->

1. Remove all `/dev` characters used in relative links and `dev/` characters in `TOC.md`
2. Move the TOC.md file to `dev`
3. Copy `dev` to `v3.0` and `v2.1`, while change the previous `v2.1` to `v2.1-legacy`
4. Remove all `aliases` lines in `dev` and `v2.1` to avoid build failure
5. Add an `_index.md` file to the root directory of `dev`/`v3.0`/`v2.1`
6. Make sure no `scripts`, `templates` or `circle.yml` files exist in `dev`/`v3.0`/`v2.1` and only keep a new version of the at the root directory of this repo.

### What is the related PR or file link(s)? <!--REMOVE this item if it is not applicable-->

https://github.com/pingcap/docs/pull/1250

### Which version does your change affect? <!--REMOVE this item if it is not applicable-->

dev, v3.0, v2.1

### Checklist <!--Check the box before the applicable item by using "- [x]"-->

- [x] Add a new file to `TOC.md`
- [x] No Tab spaces anywhere <!--Use ordinary spaces because Tab spaces can lead to CircleCI failure.-->
- [x] Leave a blank line both before and after a code block
- [x] Keep the first level heading consistent with `title` in metadata
- [x] Use *four* spaces for each level of indentation except that in `TOC.md`

sticnarf commented:
I think people might get misleaded. Now I get the latest doc even when I switch to ver 2.1. I'm not sure how much of 2.1 doc should be modified, but there is. (like transaction auto retry behavior)

lilin90 commented:
@sticnarf We're working on differentiating the three versions now. Contents that don't apply to v2.1 will be removed soon.

lilin90 commented:
@sticnarf FYI. Removed the following inapplicable content in `v2.1` (still WIP): 
- https://github.com/pingcap/docs-cn/pull/1510
- Updated transaction auto retry: https://github.com/pingcap/docs-cn/pull/1511
